code/game/objects/items/implants/implant_explosive.dm

```diff
@@ -73,25 +79,31 @@
 	message_admins("[ADMIN_LOOKUPFLW(imp_in)] has activated their [name] at [ADMIN_VERBOSEJMP(boomturf)], with cause of [cause].")
 	//If the delay is shorter or equal to the default delay, just blow up already jeez
 	if(delay <= MICROBOMB_DELAY)
-		explosion(src, devastation_range = explosion_devastate, heavy_impact_range = explosion_heavy, light_impact_range = explosion_light, flame_range = explosion_light, flash_range = explosion_light, explosion_cause = src)
-		if(imp_in)
-			imp_in.investigate_log("has been gibbed by an explosive implant.", INVESTIGATE_DEATHS)
-			imp_in.gib(TRUE)
-		qdel(src)
+		explode()
 		return
 	timed_explosion()
 
 /obj/item/implant/explosive/implant(mob/living/target, mob/user, silent = FALSE, force = FALSE)
 	for(var/target_implant in target.implants)
 		if(istype(target_implant, /obj/item/implant/explosive)) //we don't use our own type here, because macrobombs inherit this proc and need to be able to upgrade microbombs
+			var/obj/item/implant/explosive/other_implant = target_implant
+			if(other_implant.master_implant && master_implant) //we cant have two master implants at once
+				target.balloon_alert(target, "cannot fit implant!")
+				return FALSE
+			if(master_implant) //override the old implant and add in the old stats
+				explosion_devastate += other_implant.explosion_devastate
+				explosion_heavy += other_implant.explosion_heavy
+				explosion_light += other_implant.explosion_light
+				delay = min(delay + other_implant.delay, 30 SECONDS)
+				qdel(other_implant)
```

Comment:
Could you put the explosion merging into it's own proc? Then you wont have to copypaste it in the next block